lms/djangoapps/instructor/tests/test_tools.py

```diff
@@ -357,3 +359,45 @@ def msk_from_problem_urlname(course_id, urlname, block_type='problem'):
         urlname = urlname[:-4]
 
     return course_id.make_usage_key(block_type, urlname)
+
+
+@attr(shard=1)
+class TestStudentFromIdentifier(TestCase):
```

Comment:
We may want to add/update a test to verify successful retrieval of user on providing alternative identifier in cases where there is a conflict.

Reply:
By alternate identifier i mean if you were using email the alternate is username and vise versa.
Test are already here:

https://github.com/edx/edx-platform/pull/17573/files/0a65de928883805c4b3a5a7f2638e3189dfc289d#diff-e725147dad56bbfed04e1856c2d44ce2R380

https://github.com/edx/edx-platform/pull/17573/files/0a65de928883805c4b3a5a7f2638e3189dfc289d#diff-e725147dad56bbfed04e1856c2d44ce2R376

Reply:
I meant to cover the case where a user is involved in conflicting condition. As querying `duplicate_email`  with email is throwing exception, we may want to cover a case of successful retrieval of user `duplicate_email` using username.
This would assert that it is still possible to retrieve user having a conflict in username/email.

Reply:
duplicate emails are not allowed in edX platform. you can test it from registration page.
but one user can have an email for example a@e.com and other user can have a user name a@e.com. And conflict appear from here. i.e
`user = User.objects.get(Q(email=username_or_email) | Q(username=username_or_email))` will throw duplicate user error. And this is our goal.

Reply:
By `duplicate_email` I meant `cls.duplicate_email = UserFactory.create(email='foo@touchstone.com')` declaration.
This user is in conflicting condition where its email (foo@touchstone.com) is conflicting with username (foo@touchstone.com) of user `duplicate_user_name`. 
I think it would be worthy to cover a case where such conflicting users are retrieved successfully by using alternative identifier. Like querying user `duplicate_email` successfully by username and/or querying user `duplicate_user_name` successfully by email.

Reply:
Done @noraiz-anwar